dispatches/models/fossil_case/ultra_supercritical_plant/ultra_supercritical_powerplant.py

```diff
@@ -0,0 +1,1424 @@
+    tags['da_Tout'] = ("%4.2f" % (pyo.value(
+        m.fs.deaerator.mixed_state[0].temperature)))
+    tags['da_Pout'] = ("%4.1f" % (pyo.value(
+        m.fs.deaerator.outlet.pressure[0])*1e-6))
+    tags['da_Hout'] = ("%4.1f" % (pyo.value(
+        m.fs.deaerator.outlet.enth_mol[0])*1e-3))
+    tags['da_xout'] = ("%4.1f" % (pyo.value(
+        m.fs.deaerator.mixed_state[0].vapor_frac)))
+
+    ## Feed water heaters mixers
+
+    for i in m.mixer_list:
+        tags['fwh'+str(i)+'mix_steam_Fin'] = ("%4.3f" % (pyo.value(
+            m.fs.fwh_mixer[i].steam.flow_mol[0])*1e-3))
+        tags['fwh'+str(i)+'mix_steam_Tin'] = ("%4.2f" % (pyo.value(
+            m.fs.fwh_mixer[i].steam_state[0].temperature)))
+        tags['fwh'+str(i)+'mix_steam_Pin'] = ("%4.1f" % (pyo.value(
+            m.fs.fwh_mixer[i].steam.pressure[0])*1e-6))
+        tags['fwh'+str(i)+'mix_steam_Hin'] = ("%4.1f" % (pyo.value(
+            m.fs.fwh_mixer[i].steam.enth_mol[0])*1e-3))
+        tags['fwh'+str(i)+'mix_steam_xin'] = ("%4.4f" % (pyo.value(
+            m.fs.fwh_mixer[i].steam_state[0].vapor_frac)))
+        tags['fwh'+str(i)+'mix_Fout'] = ("%4.3f" % (pyo.value(
+            m.fs.fwh_mixer[i].outlet.flow_mol[0])*1e-3))
+        tags['fwh'+str(i)+'mix_Tout'] = ("%4.2f" % (pyo.value(
+            m.fs.fwh_mixer[i].mixed_state[0].temperature)))
+        tags['fwh'+str(i)+'mix_Pout'] = ("%4.1f" % (pyo.value(
+            m.fs.fwh_mixer[i].outlet.pressure[0])*1e-6))
+        tags['fwh'+str(i)+'mix_Hout'] = ("%4.1f" % (pyo.value(
+            m.fs.fwh_mixer[i].outlet.enth_mol[0])*1e-3))
+        tags['fwh'+str(i)+'mix_xout'] = ("%4.4f" % (pyo.value(
+            m.fs.fwh_mixer[i].mixed_state[0].vapor_frac)))
+
+    # BFP
+    tags['bfp_power'] = ("%4.2f" % (pyo.value(
+        m.fs.bfp.control_volume.work[0])*1e-6))
+    tags['booster_power'] = ("%4.2f" % (pyo.value(
+        m.fs.booster.control_volume.work[0])*1e-6))
+    tags['bfpt_power'] = ("%4.2f" % (pyo.value(
+        m.fs.bfpt.control_volume.work[0])*-1e-6))
+    tags['cond_power'] = ("%4.2f" % (pyo.value(
+        m.fs.cond_pump.control_volume.work[0])*1e-6))
+
+    original_svg_file = os.path.join(
+        this_file_dir(), "pfd_ultra_supercritical_pc.svg")
+    with open(original_svg_file, "r") as f:
+        svg_tag(tags, f, outfile=outfile)
+
+
+def build_plant_model(initialize_from_file=None, store_initialization=None):
+
+    # Create a flowsheet, add properties, unit models, and arcs
+    m = declare_unit_model()
+
+    # Give all the required inputs to the model
+    # Ensure that the degrees of freedom = 0 (model is complete)
+    set_model_input(m)
+    # Assert that the model has no degree of freedom at this point
+    assert degrees_of_freedom(m) == 0
+
+    set_scaling_factors(m)
+    # Initialize the model (sequencial initialization and custom routines)
+    # Ensure after the model is initialized, the degrees of freedom = 0
+    solver = initialize(m)
```

Comment:
I do not think you should be calling `initialize` here as part of the `build_plant_model` method, as this embeds initialization into model construction.

The most obvious issue with this is during the tests above - you call the `build_plant_model` method which will trigger `initialize` here as it runs. However, your first test then calls `initialize` again separately. Firstly, this is duplicated effort as you are running `initialize` twice. Secondly, this will result in unexpected test failures if `initialize` fails. - the error will appear to come from the fixture rather than the actual test, and will manifest in all the tests rather than just the initialize test.

Reply:
Thanks. This has now been updated